src/room_member.cpp

```diff
@@ -0,0 +1,203 @@
+// Copyright 2017 Tunnler Project
+// Licensed under GPLv2 or any later version
+// Refer to the license.txt file included.
+
+#include <thread>
+#include <mutex>
+
+#include "tunnler/tunnler.h"
+#include "tunnler/room.h"
+#include "tunnler/room_member.h"
+#include "tunnler/room_message_types.h"
+
+#include "tunnler/assert.h"
+
+#include "BitStream.h"
+#include "RakNetTypes.h"
+
+/// A special MAC address that tells the room we're joining to assign us a MAC address automatically.
+static const MacAddress NoPreferredMac = { 0, 0, 0, 0, 0, 0 };
+
+RoomMember::RoomMember() {
+    peer = RakNet::RakPeerInterface::GetInstance();
```

Comment:
I believe offline raknet peers do some things such as responding to pings.
If that is the case, we should consider only opening the peers once actual connection stuff is about to happen maybe?